pkg/cri/resource-manager/cache/cache.go

```diff
@@ -1110,6 +1128,47 @@ func (cch *cache) setEntry(key string, ptr, obj interface{}) error {
 	return nil
 }
 
+// checkFilePermissions checks permission bits of an existing file.
+func (cch *cache) checkFilePermissions(path string, perm, rejected os.FileMode, what string) error {
+	if info, err := os.Stat(path); err == nil && !info.IsDir() {
+		expected := perm & os.ModePerm // deliberately ignore sticky bit in comparison
+		existing := info.Mode().Perm() & os.ModePerm
+		if rejected != 0 && (existing&rejected) != 0 {
+			return cacheError("existing %s file %q has disallowed permissions set: %v",
+				what, path, existing&rejected)
+		}
+		if existing != expected && (existing|expected) != expected {
```

Comment:
nit: I think this logic is also somehow overcomplicated. Wouldn't `(existing|expected) != expected` be enough, because if that is true then certainly `existing != expected`?

Reply:
Removed extra-articulated inequality check.